spec/values/github_session_spec.rb

```diff
@@ -15,20 +26,9 @@
   let!(:client) { double(user: user_response, organization_memberships: org_response) }
   subject { GithubSession.new(cookies) }
 
-  def create_org_response(id, org, avatar_url, role)
-    double(
-      organization: double(id: id, login: org, avatar_url: avatar_url),
-      role: role
-    )
-  end
-
   before do
-    expect(BuildOctokitClient).to receive(:for)
+    allow(BuildOctokitClient).to receive(:for)
```

Comment:
Me entra la duda si lo que corresponde usar en este caso (y los de abajo) es expect o allow. Por lo que vi, la diferencia es que expect arroja error cuando el objeto no recibe el método que se indica. Entonces sería pensar si es que el hecho de que se llame o no el método afecta a la correctitud de la función testeada.
No te digo que lo cambies, yo no tengo muy claro cual sería correcto, pero te lo digo para ver que piensas tu

Reply:
No, el objetivo de esta parte es **establecer** que cuando se llame haga la llamada `BuildOctokitClient.for(...)`entonces se retorne `client`. No tiene sentido `expect` IMO.

La razón es que el módulo depende de `BuildOctokitClient`:
```
  def client
    @client ||= BuildOctokitClient.for(token: token)
  end
```
Se stubbea `BuildOctokitClient` pa hacerlo apuntar a la instancia `client` que se define arriba no más.